dbms/src/TestUtils/executorSerializer.cpp

```diff
@@ -0,0 +1,385 @@
+        traverseExecutors(dag_request, [this, &prefix](const tipb::Executor & executor) {
+            assert(executor.has_executor_id());
+            context.buf.fmtAppend("{}{}\n", prefix, executor.executor_id());
+            prefix.append(" ");
+            return true;
+        });
+        return buffer.toString();
+    }
+}
+
+// todo support more types of scalar function
+std::unordered_map<tipb::ScalarFuncSig, String> sig_to_scalar_func_name({{tipb::ScalarFuncSig::EQInt, "equals"},
+                                                                         {tipb::ScalarFuncSig::NEInt, "notEquals"},
+                                                                         {tipb::ScalarFuncSig::LogicalAnd, "and"},
+                                                                         {tipb::ScalarFuncSig::LogicalOr, "or"},
+                                                                         {tipb::ScalarFuncSig::UnaryNotInt, "not"},
+                                                                         {tipb::ScalarFuncSig::GTInt, "greater"},
+                                                                         {tipb::ScalarFuncSig::LTInt, "less"}});
+
+// todo support more types of aggregate function
+std::unordered_map<tipb::ExprType, String> sig_to_agg_func_name({{tipb::ExprType::Max, "max"}});
+
+String getJoinTypeName(tipb::JoinType tp)
+{
+    switch (tp)
+    {
+    case tipb::JoinType::TypeAntiLeftOuterSemiJoin:
+        return "AntiLeftOuterSemiJoin";
+    case tipb::JoinType::TypeLeftOuterJoin:
+        return "LeftOuterJoin";
+    case tipb::JoinType::TypeRightOuterJoin:
+        return "RightOuterJoin";
+    case tipb::JoinType::TypeLeftOuterSemiJoin:
+        return "LeftOuterSemiJoin";
+    case tipb::JoinType::TypeAntiSemiJoin:
+        return "AntiSemiJoin";
+    case tipb::JoinType::TypeInnerJoin:
+        return "InnerJoin";
+    case tipb::JoinType::TypeSemiJoin:
+        return "SemiJoin";
+    default:
+        throw TiFlashException(fmt::format("Not supported Join type {}", tp), Errors::Coprocessor::Internal);
+    }
+}
+
+String getJoinExecTypeName(tipb::JoinExecType tp)
+{
+    switch (tp)
+    {
+    case tipb::JoinExecType::TypeHashJoin:
+        return "HashJoin";
+    default:
+        throw TiFlashException(fmt::format("Not supported Join exectution type {}", tp), Errors::Coprocessor::Internal);
+    }
+}
+
+String getExchangeTypeName(tipb::ExchangeType tp)
+{
+    switch (tp)
+    {
+    case tipb::ExchangeType::Broadcast:
+        return "Broadcast";
+    case tipb::ExchangeType::PassThrough:
+        return "PassThrough";
+    case tipb::ExchangeType::Hash:
+        return "Hash";
+    default:
+        throw TiFlashException(fmt::format("Not supported Exchange type :{}", tp), Errors::Coprocessor::Internal);
+    }
+}
+
+String getFieldTypeName(Int32 tp)
+{
+    switch (tp)
+    {
+    case TiDB::TypeTiny:
+        return "tiny";
+    case TiDB::TypeShort:
+        return "short";
```

Comment:
maybe `Short` and `Tiny`? And can use [M Macros](https://stackoverflow.com/a/49595815) to get enum item name from its value, just like https://github.com/pingcap/tiflash/blob/master/dbms/src/Storages/Transaction/TiDB.h#L59-L87